pkg/api/types.go

```diff
@@ -55,6 +55,7 @@ type Properties struct {
 	AADProfile              *AADProfile              `json:"aadProfile,omitempty"`
 	CustomProfile           *CustomProfile           `json:"customProfile,omitempty"`
 	HostedMasterProfile     *HostedMasterProfile     `json:"hostedMasterProfile,omitempty"`
+	UpgradeProfile          *UpgradeContainerService `json:"upgradeProfile,omitempty"`
```

Comment:
I don't understand the purpose of this change.  Plus it's not the correct way to represent old and new versions. Can you elaborate?

Reply:
reverted as discussed